### PR TITLE
Endpoint-registry rework

### PR DESCRIPTION
**Pull request by akiskips**
Rework of the Endpoint Registry Service to address new API. Main changes:
- new API
- ORM approach (https://typeorm.io/#/)
- Endpoind-Resolver service updated to the new API
- Initialiser and docker updates

Note: Logging branch was merged into this one

### CODE REVIEW

File: src/ts/cmd/endpoint-registry/package.json [hunk 1]
```diff
@@ -4,13 +4,14 @@
   "description": "",
   "main": "src/index.js",
   "scripts": {
-    "test": "mocha --require ts-node/register --exit test/unit/**/*.spec.ts",
+    "dev": "tsc-watch --onSuccess \"node ./dist/server.js\"",
+    "test": "npm run test-integration-only",
```
Comment (etoster) [inline src/ts/cmd/endpoint-registry/package.json:8]: Are unit tests omitted on purpose here? Are they not there yet?

Comment (kasimrehman) [reply src/ts/cmd/endpoint-registry/package.json:8]: "test-unit-only" has been added below. There are no unit tests at the moment, so does not make sense to include them in the default.

File: docs/postman/sample_requests/sample_interaction.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
           {
             "idType": "URI",
             "type": "GlobalReference",
-            "value": "opcfoundation.org/specifications-unified-architecture/part-100-device-information-model",
+            "value": "opcfoundation.org/specifications-unified-architecture/part-100-device-information-model/10JF-1234-Jf14-PP22",
```
Comment (Manu3756) [inline docs/postman/sample_requests/sample_interaction.json:28]: remove /10JF-1234-Jf14-PP2
the semanticId is the same for each submodel instance.

File: src/ts/cmd/endpoint-registry/src/middleware/checks.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  if (submodelsArray && submodelsArray.length > 0) {
+    submodelsArray.forEach(submodel => {
+      //based on this id the routing towards the storage adapter takes place
+        let submodelID:string = submodel.idShort;
+        logger.debug("id " + submodelID);
+    if(!submodelID) {
+        logger.error("Missing id in submodel "+submodel.identification.id);
+        throw new HTTP422Error("Missing required fields in Request: idShort");
+
+      }
```
Comment (Manu3756) [inline src/ts/cmd/endpoint-registry/src/middleware/checks.ts:40]: idShort is not required as far as I know. submodel.identification.id should be the main routing key.

File: src/ts/cmd/endpoint-registry/src/services/registry/daos/entities/EndpointEntity.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import {Entity, Column, PrimaryGeneratedColumn, Unique, ManyToOne, Index, PrimaryColumn, JoinTable} from "typeorm";
+import { AASDescriptorEntity } from "./AASDescriptorEntity";
+
+@Entity()
+//the combination for address and type should also be unique (to avoid saving double)
+
+export class EndpointEntity {
+
+  @PrimaryColumn({
+    length: 1024
```
Comment (Manu3756) [inline src/ts/cmd/endpoint-registry/src/services/registry/daos/entities/EndpointEntity.ts:10]: are 1024 chars enough?

File: src/ts/cmd/endpoint-registry/src/utils/httpErrors.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
+
```
Comment (Manu3756) [inline src/ts/cmd/endpoint-registry/src/utils/httpErrors.ts:1]: Maybe a module is the better way to go. Not sure if we should write all this classes by our own. 
https://www.npmjs.com/package/http-errors

File: docs/postman/AAS.postman_collection.json [hunk 6]
```diff
@@ -171,50 +367,137 @@
...
+								{
+									"key": "REGISTRY_ADMIN_PASSWORD",
+									"value": "{{pwd}}",
+									"disabled": true
+								}
+							],
+							"variable": [
+								{
+									"key": "aasId",
+									"value": "www.sampleOperator/aas"
```
Comment (etoster) [inline docs/postman/AAS.postman_collection.json:499]: Escape special character "/" here: "www.sampleOperator%2Faas"

File: src/ts/cmd/endpoint-registry/package.json (hunk 1 above)
Comment (etoster) [reply src/ts/cmd/endpoint-registry/package.json:8]: Still, the integration almost completely fail. See https://travis-ci.com/github/SAP/i40-aas/jobs/320927566#L1068

Comment (kasimrehman) [reply src/ts/cmd/endpoint-registry/package.json:8]: Yes, this is because we as a team decided that "done criteria" do not apply for this backlog item.

Comment (etoster) [reply src/ts/cmd/endpoint-registry/package.json:8]: Was just checking that this behavior is intended.

Comment (akiskips) [reply src/ts/cmd/endpoint-registry/package.json:8]: Tests for the fix part of the Registry API are now in place

File: src/ts/cmd/endpoint-registry/src/middleware/checks.ts (hunk 3 above)
Comment (akiskips) [reply src/ts/cmd/endpoint-registry/src/middleware/checks.ts:40]: updated the validation to check for AASDescriptor specific fields

File: src/ts/cmd/endpoint-registry/src/utils/httpErrors.ts (hunk 5 above)
Comment (akiskips) [reply src/ts/cmd/endpoint-registry/src/utils/httpErrors.ts:1]: Indeed we can consider using something like this. We started initially with our own to have more control on error handling and reporting. I will make a issue in jira

File: docs/postman/AAS.postman_collection.json (hunk 6 above)
Comment (akiskips) [reply docs/postman/AAS.postman_collection.json:499]: Right, corrected

File: docs/postman/sample_requests/sample_interaction.json (hunk 2 above)
Comment (akiskips) [reply docs/postman/sample_requests/sample_interaction.json]: done

File: docs/postman/AAS.postman_collection.json (hunk 6 above)
Comment (etoster) [reply docs/postman/AAS.postman_collection.json:499]: It's not corrected on the current commit. Did you upload the up-do-date Postman collection?

**Pull request merged**